### PR TITLE
os/binfmt/libelf: Correctly assign number of blocks for caching

### PR DESCRIPTION
**Pull request by Nishtha97**
-Assign no. of blocks for caching based on correctly updated number of blocks in file

Signed-off-by: Nishtha97 <n.maloo@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [a0d4e123c55925439eaaf34c9c011192fede25c2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a0d4e123c55925439eaaf34c9c011192fede25c2] - Code Rule Check OK.

File: os/binfmt/libelf/libelf_cache.c [hunk 1]
```diff
@@ -447,23 +447,22 @@ int elf_cache_init(int filfd, uint16_t offset, off_t filelen, uint8_t compressio
...
 
 	/* Initialize the ELF params */
-	number_blocks_caching = CONFIG_ELF_CACHE_BLOCKS_COUNT;
 	cache_blocks_size = CONFIG_ELF_CACHE_BLOCK_SIZE;
 	file_len = filelen;
 	number_of_blocks = file_len / cache_blocks_size;
 	elf_compress_type = compression_type;
 
+	/* Extra unaligned data apart from blocksize */
+	if (file_len % cache_blocks_size)
```
Comment (sunghan-chang) [inline os/binfmt/libelf/libelf_cache.c]: Needs brackets

**Comment by sunghan-chang** [comment]
@Nishtha97 Could you let me know the test result? Why is this necessary even yesterday it was modified?

File: os/binfmt/libelf/libelf_cache.c (hunk 1 above)
Comment (sijupunnnoose) [reply os/binfmt/libelf/libelf_cache.c]: > Is this missing part of [2b7c427](https://github.com/Samsung/TizenRT/commit/2b7c427cc42bc63113caf3d8e25b0dfc4d73fce5)?

Yes, it was a mistake from engineer end, round off 10% was not taken care.

**Comment by sijupunnnoose** [comment]
> @Nishtha97 Could you let me know the test result? Why is this necessary even yesterday it was modified?

It was part of the previous patch, but missed by mistake, 10% of binary size can 56.87 blocks, in this case 56.86 should be rounded to 57 blocks.

**Comment by seinfra** [comment]
Target : [fb61a6013ef8a635a2c923b61397345e67e2d827] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [fb61a6013ef8a635a2c923b61397345e67e2d827] - Code Rule Check (C++) OK.

**Comment by sunghan-chang** [comment]
@sijupunnnoose @Nishtha97 Could I merge this?

**Pull request merged**